### PR TITLE
Improve the UI of the marine vendors

### PR DESCRIPTION
**Pull request by psykzz**
## About The Pull Request
Improve the UI of the marine vendors

## Why It's Good For The Game

QOL on the UI and allows for more items to be added without making the ui confusing.

## Changelog
:cl:
tweak: tweaked the UIfothe marine vendors.
/:cl:

image
image

### CODE REVIEW

File: code/game/objects/machinery.dm [hunk 1]
```diff
@@ -153,10 +153,11 @@
 	. = ..()
 	if(.)
 		return
+
 	if(!is_operational())
 		return TRUE
-
-	if(usr.restrained() || usr.lying || usr.stat != CONSCIOUS)
```
Comment (psykzz) [inline code/game/objects/machinery.dm]: This is removing duplicate code all over from marine vendor down to the roots.

File: code/modules/mob/mob_helpers.dm [hunk 2]
```diff
@@ -358,7 +358,7 @@ It's fairly easy to fix if dealing with single letters but not so much with comp
 
 
 /mob/proc/incapacitated(ignore_restrained)
-	return (stat || (!ignore_restrained && restrained()))
+	return (stat == UNCONSCIOUS || (!ignore_restrained && restrained()))
```
Comment (DominikPanic) [inline code/modules/mob/mob_helpers.dm]:
```suggestion
	return (stat != CONSCIOUS || (!ignore_restrained && restrained()))
```

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 3]
```diff
@@ -250,7 +306,8 @@
 
 			var/type_p = L[3]
 
-			new type_p(loc)
+			var/obj/item = new type_p(loc)
+			usr.put_in_any_hand_if_possible(item)
```
Comment (DominikPanic) [inline code/game/objects/machinery/vending/new_marine_vendors.dm]: Use put_in_hands, this abomination should be removed

**Comment by Rohesie** [comment]
```
[03:32:27] Runtime in inventory.dm, line 74: undefined proc or verb /obj/effect/essentials_set/basic_specialist/dropped(). 
proc name: put in hands (/mob/proc/put_in_hands)
usr: CoreFlare/(Todo 'Nerd' Hayabushi)
usr.loc: (Canterbury (29, 25, 4))
src: Todo \'Nerd\' Hayabushi (/mob/living/carbon/human)
src.loc: the floor (29,25,4) (/turf/open/floor/mainship)
call stack:
Todo \'Nerd\' Hayabushi (/mob/living/carbon/human): put in hands(the basic specialist (/obj/effect/essentials_set/basic_specialist))
GHMME Automated Specialist Clo... (/obj/machinery/marine_selector/clothes/specialist/alpha): Topic("src=\[0x2100f52f];vend=2", /list (/list), /datum/topic_state/default (/datum/topic_state/default))
/datum/nanoui (/datum/nanoui): Topic("src=\[0x2100f52f];vend=2", /list (/list))
CoreFlare (/client): Topic("src=\[0x2100f52f];vend=2", /list (/list), /datum/nanoui (/datum/nanoui))
CoreFlare (/client): Topic("src=\[0x2100f52f];vend=2", /list (/list), /datum/nanoui (/datum/nanoui))
```

**Comment by psykzz** [comment]
Fixed @Rohesie

**Comment by Rohesie** [comment]
```
[20:02:34] Runtime in inventory.dm, line 74: undefined proc or verb /obj/effect/essentials_set/basic_specialist/dropped(). 
proc name: put in hands (/mob/proc/put_in_hands)
usr: Ondrej008/(Alejandro Olphert)
usr.loc: (Squad Delta Preparation (168, 35, 3))
src: Alejandro Olphert (/mob/living/carbon/human)
src.loc: the catwalk (168,35,3) (/turf/open/floor/plating/plating_catwalk)
call stack:
Alejandro Olphert (/mob/living/carbon/human): put in hands(the basic specialist (/obj/effect/essentials_set/basic_specialist))
GHMME Automated Specialist Clo... (/obj/machinery/marine_selector/clothes/specialist/delta): Topic("src=\[0x2100e861];vend=2", /list (/list), /datum/topic_state/default (/datum/topic_state/default))
/datum/nanoui (/datum/nanoui): Topic("src=\[0x2100e861];vend=2", /list (/list))
Ondrej008 (/client): Topic("src=\[0x2100e861];vend=2", /list (/list), /datum/nanoui (/datum/nanoui))
Ondrej008 (/client): Topic("src=\[0x2100e861];vend=2", /list (/list), /datum/nanoui (/datum/nanoui))
```

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 4]
```diff
@@ -250,7 +306,10 @@
 
 			var/type_p = L[3]
 
-			new type_p(loc)
+			var/item = new type_p(loc)
+			if(isitem(item))
```
Comment (Rohesie) [inline code/game/objects/machinery/vending/new_marine_vendors.dm]: Might want to add this as a variable of the product, if it's meant to be placed in hands or not.

**Comment by Rohesie** [comment]
Static datums might offer cleaner code and better performance, but this is a direct improvement on the previous code.

**Comment by psykzz** [comment]
@Rohesie Agree with your previous points about the static datum, but im not sure how to do them right.

I tried a little in another PR i believe, but it didn't go too well so prefer to merge this and tackle it later.

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 5]
```diff
@@ -250,7 +306,10 @@
...
+				var/obj/item/O = item
+				usr.put_in_hands(O)
```
Comment (DominikPanic) [inline code/game/objects/machinery/vending/new_marine_vendors.dm]: Personally I disagree with the putting in hand feature, would you be alright with asking the players and going by what they prefer the most?

**Comment by Rohesie** [comment]
All good on the static datum. I was saying it'd be ideal, but let's not get too perfectionist here.
As said, this is a direct improvement.

File: code/game/objects/machinery/vending/new_marine_vendors.dm (hunk 5 above)
Comment (spookydonut) [reply code/game/objects/machinery/vending/new_marine_vendors.dm]: Good change, having to click twice for each item to vend + pickup THEN equip it is annoying.

File: code/game/objects/machinery/vending/new_marine_vendors.dm (hunk 4 above)
Comment (spookydonut) [reply code/game/objects/machinery/vending/new_marine_vendors.dm]: Are there any things you can vend that aren't items?  You can only pickup items with your hands.

File: code/modules/mob/mob_helpers.dm [hunk 6]
```diff
@@ -358,7 +358,7 @@ It's fairly easy to fix if dealing with single letters but not so much with comp
 
 
 /mob/proc/incapacitated(ignore_restrained)
-	return (stat || (!ignore_restrained && restrained()))
+	return (stat != CONSCIOUS || (!ignore_restrained && restrained()))
```
Comment (spookydonut) [inline code/modules/mob/mob_helpers.dm]: don't really need to change this but sure +1 clarity

File: nano/templates/marine_selector.tmpl [hunk 7]
```diff
@@ -1,35 +1,135 @@
+<style>
```
Comment (spookydonut) [inline nano/templates/marine_selector.tmpl]: are the PR description screenshot(s) this template?

Comment (psykzz) [reply nano/templates/marine_selector.tmpl]: Yeah

File: code/game/objects/machinery/vending/new_marine_vendors.dm (hunk 4 above)
Comment (psykzz) [reply code/game/objects/machinery/vending/new_marine_vendors.dm]: There are some special effect types that spawn the crates.

File: nano/templates/marine_selector.tmpl (hunk 7 above)
Comment (spookydonut) [reply nano/templates/marine_selector.tmpl]: looks good

File: code/game/objects/machinery/vending/new_marine_vendors.dm (hunk 4 above)
Comment (spookydonut) [reply code/game/objects/machinery/vending/new_marine_vendors.dm]: The crates are still obj/item though?

Comment (psykzz) [reply code/game/objects/machinery/vending/new_marine_vendors.dm]: there are `obj/effect/essentials_set/leader` that when spawned, create the crate.

File: tools/midi2piano/midi2piano/obj/x86/Debug/midi2piano.csproj.CoreCompileInputs.cache [hunk 8]
```diff
@@ -0,0 +1 @@
+d9f161e9d0130fbebe72002a19544a2a8fae236d
```
Comment (Rohesie) [inline tools/midi2piano/midi2piano/obj/x86/Debug/midi2piano.csproj.CoreCompileInputs.cache]: Is this an intended change?

Comment (spookydonut) [reply tools/midi2piano/midi2piano/obj/x86/Debug/midi2piano.csproj.CoreCompileInputs.cache]: this file should be in .gitignore lol

Comment (psykzz) [reply tools/midi2piano/midi2piano/obj/x86/Debug/midi2piano.csproj.CoreCompileInputs.cache]: they are not covered by my gitignore :|
```
#ignore misc BYOND files
Thumbs.db
*.log
*.int
*.rsc
*.dmb
*.lk
*.backup
*.dmm.before
data/
cfg/
tmp/
tools/Runtime Condenser/Input.txt
tools/Runtime Condenser/Output.txt
tools/MapDaemon/logs/*.txt
.vscode/*
```

Comment (spookydonut) [reply tools/midi2piano/midi2piano/obj/x86/Debug/midi2piano.csproj.CoreCompileInputs.cache]: I meant they should be but weren't so I added it in a different pr

**Comment by spookydonut** [comment]
@psykzz merge conflict

**Comment by Rohesie** [comment]
`mixed <tab><space> indentation detected` from the linter

**Comment by psykzz** [comment]
Yeah i just can't actually find them :|

**Comment by Rohesie** [comment]
Vendors failed to work when tested.
Could you test it locally?

**Comment by psykzz** [comment]
yeah im guessing a conflict caused an issue as people are changing loadouts daily

**Comment by psykzz** [comment]
@Rohesie vendor should be fixed now there was a small typo that meant you could get infinite gear.

**Comment by Rohesie** [comment]
`mixed <tab><space> indentation detected`

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 9]
```diff
@@ -320,53 +393,53 @@
...
+			list("Shotgun scabbard", 0, /obj/item/storage/large_holster/m37, MARINE_CAN_BUY_BACKPACK, "standard"),
+			list("Machete scabbard", 0, /obj/item/storage/large_holster/machete/full, MARINE_CAN_BUY_BACKPACK, "standard"),
+			list("Welderpack", 0, /obj/item/storage/backpack/marine/engineerpack, MARINE_CAN_BUY_BACKPACK, "standard"),
+		list("utility", "header", "WEBBING", 1, ""),
+			list("Tactical Black Vest", 0, /obj/item/clothing/tie/storage/black_vest, MARINE_CAN_BUY_WEBBING, "suggested"),
+			list("Tactical Webbing", 0, /obj/item/clothing/tie/storage/webbing, MARINE_CAN_BUY_WEBBING, "standard"),
+			list("Shoulder Handgun Holster", 0, /obj/item/clothing/tie/holster, MARINE_CAN_BUY_WEBBING, "standard"),
+		list("utility", "header", "BELT", 1, ""),
+			list("Tool belt", 0, /obj/item/storage/belt/utility/full, MARINE_CAN_BUY_BELT, "suggested"),
+      list("Belt Harness", 0, /obj/item/belt_harness/marine, MARINE_CAN_BUY_BELT, "standard"),
```
Comment (spookydonut) [inline code/game/objects/machinery/vending/new_marine_vendors.dm]:
```suggestion
		list("Belt Harness", 0, /obj/item/belt_harness/marine, MARINE_CAN_BUY_BELT, "standard"),
```

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 10]
```diff
@@ -517,40 +585,44 @@
...
+
+		list("utility", "header", "BELT", 1, ""),
+			list("Standard ammo belt", 0, /obj/item/storage/belt/marine, MARINE_CAN_BUY_BELT, "suggested"),
+			list("M39 holster belt", 0, /obj/item/storage/large_holster/m39, MARINE_CAN_BUY_BELT, "standard"),
+			list("Shotgun ammo belt", 0, /obj/item/storage/belt/shotgun, MARINE_CAN_BUY_BELT, "standard"),
+			list("Knives belt", 0, /obj/item/storage/belt/knifepouch, MARINE_CAN_BUY_BELT, "standard"),
+			list("Pistol belt", 0, /obj/item/storage/belt/gun/m4a3, MARINE_CAN_BUY_BELT, "standard"),
+			list("Revolver belt", 0, /obj/item/storage/belt/gun/m44, MARINE_CAN_BUY_BELT, "standard"),
+			list("G8 general utility pouch", 0, /obj/item/storage/belt/sparepouch, MARINE_CAN_BUY_BELT, "standard"),
+      list("Belt Harness", 0, /obj/item/belt_harness/marine, MARINE_CAN_BUY_BELT, "standard"),
```
Comment (spookydonut) [inline code/game/objects/machinery/vending/new_marine_vendors.dm]:
```suggestion
		list("Belt Harness", 0, /obj/item/belt_harness/marine, MARINE_CAN_BUY_BELT, "standard"),
```

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 11]
```diff
@@ -578,44 +650,43 @@
...
+			list("Shoulder Handgun Holster", 0, /obj/item/clothing/tie/holster, MARINE_CAN_BUY_WEBBING, "standard"),
+		list("utility", "header", "BELT", 1, ""),
+			list("Standard ammo belt", 0, /obj/item/storage/belt/marine, MARINE_CAN_BUY_BELT, "standard"),
+			list("Shotgun ammo belt", 0, /obj/item/storage/belt/shotgun, MARINE_CAN_BUY_BELT, "standard"),
+			list("M39 holster belt", 0, /obj/item/storage/large_holster/m39, MARINE_CAN_BUY_BELT, "standard"),
+			list("Knives belt", 0, /obj/item/storage/belt/knifepouch, MARINE_CAN_BUY_BELT, "standard"),
+			list("Pistol belt", 0, /obj/item/storage/belt/gun/m4a3, MARINE_CAN_BUY_BELT, "standard"),
+			list("Revolver belt", 0, /obj/item/storage/belt/gun/m44, MARINE_CAN_BUY_BELT, "standard"),
+			list("G8 general utility pouch", 0, /obj/item/storage/belt/sparepouch, MARINE_CAN_BUY_BELT, "standard"),
+      list("Belt Harness", 0, /obj/item/belt_harness/marine, MARINE_CAN_BUY_BELT, "standard"),
```
Comment (spookydonut) [inline code/game/objects/machinery/vending/new_marine_vendors.dm]:
```suggestion
		list("Belt Harness", 0, /obj/item/belt_harness/marine, MARINE_CAN_BUY_BELT, "standard"),
```

**Comment by spookydonut** [comment]
Obsoleted by #3120

**Pull request closed without merging**